connect-inject/handler.go

```diff
@@ -607,14 +607,14 @@ func (h *Handler) serviceMetricsPort(pod *corev1.Pod) (string, error) {
 	// If that has been set, it'll be used as the port for getting service
 	// metrics as well, unless overridden by the service-metrics-port annotation.
 	if raw, ok := pod.Annotations[annotationPort]; ok && raw != "" {
-		return determineAndValidatePort(pod, annotationServiceMetricsPort, raw)
+		return determineAndValidatePort(pod, annotationServiceMetricsPort, raw, true)
```

Comment:
Does it make sense to add a text snippet here describing why the serviceMetricsPort can use a privileged port but the scrape port and merged metrics port cannot? It might be tricky to understand the rationale behind doing so in the absence of the context of Luke's PR comment.

Reply:
Totally! Added this.